addOns/pscanrules/src/test/java/org/zaproxy/zap/extension/pscanrules/TestInfoSessionIdURLUnitTest.java

```diff
@@ -0,0 +1,427 @@
+/*
+ * Zed Attack Proxy (ZAP) and its related class files.
+ *
+ * ZAP is an HTTP/HTTPS proxy for assessing web application security.
+ *
+ * Copyright 2019 The ZAP Development Team
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.zaproxy.zap.extension.pscanrules;
+
+import static org.hamcrest.Matchers.empty;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.not;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertThat;
+
+import java.util.ArrayList;
+import java.util.List;
+import net.htmlparser.jericho.Source;
+import org.apache.commons.httpclient.URI;
+import org.apache.commons.httpclient.URIException;
+import org.junit.Test;
+import org.parosproxy.paros.model.Model;
+import org.parosproxy.paros.model.OptionsParam;
+import org.parosproxy.paros.network.HttpMalformedHeaderException;
+import org.parosproxy.paros.network.HttpMessage;
+import org.parosproxy.paros.network.HttpRequestHeader;
+import org.zaproxy.zap.extension.httpsessions.HttpSessionsParam;
+import org.zaproxy.zap.utils.ZapXmlConfiguration;
+
+public class TestInfoSessionIdURLUnitTest extends PassiveScannerTest<TestInfoSessionIdURL> {
+
+    private HttpMessage msg;
+    private static final String BODY = "Some text in the response, doesn't matter.\nLine 2\n";
+
+    @Override
+    protected TestInfoSessionIdURL createScanner() {
+
+        TestInfoSessionIdURL scanner = new TestInfoSessionIdURL();
+        //        scanner.setConfig(new ZapXmlConfiguration());
+        setUpHttpSessionsParam();
+        return scanner;
+    }
+
+    protected HttpMessage createHttpMessageWithRespBody(String responseBody)
+            throws HttpMalformedHeaderException, URIException {
+
+        HttpRequestHeader requestHeader = new HttpRequestHeader();
+        requestHeader.setURI(new URI("http://example.com", false));
+
+        msg = new HttpMessage();
+        msg.setRequestHeader(requestHeader);
+        msg.setResponseBody(responseBody);
+        msg.setResponseHeader(
+                "HTTP/1.1 200 OK\r\n"
+                        + "Server: Apache-Coyote/1.1\r\n"
+                        + "Content-Type: text/plain\r\n"
+                        + "Content-Length: "
+                        + responseBody.length()
+                        + "\r\n");
+        return msg;
+    }
+
+    @Test
+    public void shouldHaveSessionIdsInConfig() throws Exception {
```

Comment:
This shouldn't be a test (it's not testing the scanner), if you really want to assert it should be done when setting up the test instead (e.g. `createScanner` or `setUpHttpSessionsParam`), although the `HttpSessionsParam` used for the tests should use hardcoded session IDs (e.g. mock and return the required IDs) to not rely on the default ones (while not likely to change for the tests it shouldn't matter what the default sessions IDs are).

Reply:
Can you explain how to mock and return the required IDs as I don't think I've done that before.

Reply:
You would have to use Mockito/PowerMock, but since the `HttpSessionsParam` is already being set up in `setUpHttpSessionsParam` you can just set the default tokens there, e.g.:
```Java
httpSessions.setDefaultTokens(Arrays.asList(new HttpSessionToken("jsessionid"), new HttpSessionToken("cfid")));
```
and this test method can be removed.

Reply:
I removed that test method. However, when I inserted 
>httpSessions.setDefaultTokens(Arrays.asList(new HttpSessionToken("jsessionid"), new HttpSessionToken("cfid")));
into setUpHttpSessionsParam "httpSessions" could not be resolved. I tried fixing this by changing it to HttpSession, but there was no setDefaultTokens function for that class. Do you have an idea on how to fix this? Thanks for the help.

Reply:
Sorry, the `httpSessions` is the `HttpSessionsParam` previosuly added to `OptionsParam`.

Reply:
Oh okay, I think I understand what you're saying. You mean something like this?
```
        HttpSessionsParam httpSessions = new HttpSessionsParam();
        options.addParamSet(httpSessions);
        httpSessions.setDefaultTokens(Arrays.asList(new HttpSessionToken("jsessionid"), new HttpSessionToken("cfid")));
```

Reply:
Yes.